### DISCUSSION TITLE
Is table2geo in some way automatic?

### DISCUSSION THREAD

**Discussion by aborruso** [neocarto/bertin#116] [Q&A] [answered]
Hi,
I have added a CSV in [my OBS notebook](https://observablehq.com/d/fa939367237d9243) using `d3`

```
citta = d3.csv(citta_url, d3.autoType)
```

The csv URL is <https://raw.githubusercontent.com/ondata/ondate-calore/main/data/ondate-calore_oggi.csv>.

Then I have added it to my Bertin map simply calling it, without using table2geo. So it seems that Bertin does the conversion to geojson automatically.
Am I right?  If yes, when does it happen? Only when there are two columns named "latitude" and "longitude"?
If this isn't documented, I'd add, it's very useful and convenient to know.

Thank you again

```
    {
      type: "layer",
      geojson: citta,
      fill: "#1f0754",
      stroke: "#1f0754",
      strokeWidth: 0.5,
      symbol: "circle",
      symbol_size: 100,
      tooltip: ["$citta"],
      fill: {
        type: "typo",
        values: "livello",
        colors: ["#5bd601", "#e8d203", "#ff7f02", "#c40001"],
        order: ["Livello0", "Livello1", "Livello2", "Livello3"]
      }
    }
```

**Comment by neocarto** [comment, marked as answer]
Yes. It works automatically with the fields  "lon",  "Lon",  "LON", "lng", "Lng",  "LNG", "Longitude", "longitude", "lat", "Lat", "LAT", "Latitude", "latitude"

**Reply by aborruso** [reply]
Thank you very much
